cpp/src/arrow/ipc/options.h

```diff
@@ -66,6 +67,10 @@ struct ARROW_EXPORT IpcWriteOptions {
   /// like compression
   bool use_threads = true;
 
+  /// \brief Format version to use for IPC messages and their
+  /// metadata. Presently using V4 version (readable by v0.8.0 and later).
+  MetadataVersion metadata_version = MetadataVersion::V4;
```

Comment:
I find it a bit weird to expose this an option. Will we be able to write data compatible with other metadata versions?

Reply:
Yes. This is what I've been saying in the V4/V5 MetadataVersion discussion e-mail

Reply:
Oops, I had misunderstood that.

Reply:
see https://issues.apache.org/jira/browse/ARROW-9265